package.json

```diff
@@ -36,7 +36,8 @@
     "tsup": "^8.3.6",
     "use-mouse-matrix-transform": "^1.3.0",
     "vite": "^6.0.11",
-    "vite-tsconfig-paths": "^5.1.4"
+    "vite-tsconfig-paths": "^5.1.4",
+    "@datastructures-js/priority-queue": "^6.3.2"
```

Comment:
This seems to be an often used package for priority queues, but let me know if you prefer to use a local variant of the datastructure to reduce the dependencies.